src/setup_payload/AdditionalDataPayloadGenerator.cpp

```diff
@@ -53,7 +53,7 @@ AdditionalDataPayloadGenerator::generateAdditionalDataPayload(uint16_t lifetimeC
     size_t rotatingDeviceIdBufferSize = 0;
 
     // Initialize TLVWriter
-    writer.Init(chip::System::PacketBufferHandle::New(chip::System::PacketBuffer::kMaxSize));
+    writer.Init(chip::System::PacketBufferHandle::New(RotatingDeviceId::kRotatingDeviceIdMaxLength));
```

Comment:
This was not tested. The TLV writer has overhead requirements, but `kRotatingDeviceIdMaxLength` is:
`static constexpr unsigned kRotatingDeviceIdMaxLength = kRotatingDeviceIdCounterStringMaxLength + kRotatingDeviceIdHashSuffixLength;`
--> It does not account for any TLV container overhead and buffer would be incomplete (and return error).

Unit tests would catch invalid assumptions about sizing.

Overall, I think the kMaxSize (which is also seen elsewhere in codebase) was about not having to care about the size, and just getting the max possible, which is very bad practice when memory is at a premium and the max size is large.

Suggest finding (without going too deep or making too brittle) a compromise size by:

1- Separating generation of payloads as a first step
2- Using the expected max sizes of all generated payloads as a starting point for sizing (e.g. RotatingDeviceId::kRotatingDeviceIdMaxLength)
3- Adding a fixed constant TLV overhead of 32-64 bytes with a comment that this may need to be updated if this payload becomes much bigger.

BTW, this is something we will eventually have to reckon with and is not unique to your PR, but this is one case among many.